File: examples/fibonacci/src/main/vivado_hls/fib.c
long long fib(int n)
{
  long long curr = 0;
  long long next = 1;
  int i;
  for(i = 0; i < n; i++) {
    long long prev = curr;
    curr = next;
    next += prev;
  }
  return curr;
}

File: examples/fibonacci/src/main/fortrock/fib.f
      SUBROUTINE FIB(N,O)
      INTEGER N,O,I,CURR,NEXT,PREV
      CURR = 0
      NEXT = 1
      I    = 0
 0001 CONTINUE
      IF(I >= N)THEN
        GOTO 0002
      ENDIF
      PREV = CURR
      CURR = NEXT
      NEXT = NEXT + PREV
      I = I + 1
      GOTO 0001
 0002 CONTINUE
      O = CURR
      RETURN
      END
